kong/concurrency.lua

```diff
@@ -91,7 +89,7 @@ function concurrency.with_coroutine_mutex(opts, fn)
     error("invalid value for opts.on_timeout", 2)
   end
 
-  if get_phase() == "init_worker" then
+  if not check_phase_yieldable() then
```

Comment:
`in_yieldable_phase()` would be clearer.  I understand that there is an optional `phase` argument, but as we're not going to supply that in most cases, I'd rather optimize the name for the standard use case.

Reply:
Function renamed!